src/rt/sched/behaviourcore.h

```diff
@@ -519,29 +519,34 @@ namespace verona::rt
         Cown::acquire(cown);
     }
 
-    static std::tuple<size_t, size_t>
-    handle_read_only_enqueue(Slot* prev_slot, Slot* new_slot, Cown* cown)
+    static std::tuple<size_t, size_t> handle_read_only_enqueue(
+      Slot* prev_slot,
+      Slot* chain_first_slot,
+      Slot* chain_last_slot,
+      bool all_reads,
+      size_t first_consecutive_readers_count,
+      Cown* cown)
     {
       size_t ref_count = 0, ex_count = 0;
       bool first_reader;
 
-      if (prev_slot && (prev_slot->set_next_slot_reader(new_slot)))
+      if (prev_slot && (prev_slot->set_next_slot_reader(chain_first_slot)))
       {
         Logging::cout() << " Previous slot is a writer or blocked reader cown "
-                        << *new_slot << Logging::endl;
+                        << *chain_first_slot << Logging::endl;
         yield();
         goto fn_out;
```

Comment:
I wonder if direct returns in this code would be clearer. 

```suggestion
        return {0, 0};
```

I'm also not sure I get the usage of `ex_count` here.  This is that all the reads in this chain have seen a `READ_AVAILABLE`, so they can all decrease there execution count by 1?